### DISCUSSION TITLE
Best way to filter for elements that yield false?

### DISCUSSION THREAD

**Discussion by dbofmmbt** [nushell/nushell#9919] [Q&A] [answered]
I'm doing some text processing and wanted to remove lines which are not empty. I came up with this:

```nu
['', "abc"] | filter { is-empty | not $in }
```

Is `| not $in` the simplest way to negate a boolean coming from the pipeline?

This is related to https://github.com/nushell/nushell/discussions/7275. Seeing the example code there, I guess `not` is very flexible because you call deal with structured data too. But maybe there's a shortcut I don't know about.

Edit: I changed the question title because there was a bit of "X-Y problem" on it.

**Comment by fdncred** [comment]
to get rid of empty things you can play around with `compact`

**Reply by dbofmmbt** [reply]
`["", "abc"] | compact` yielded me two entries. I guess it just removes null, but keeps empty strings.

**Comment by fdncred** [comment]
you can also play with `not-in`

**Comment by panicbit** [comment, marked as answer]
Another way to phrase this using `where` instead of `filter` could be:
```nu
['', "abc"] | where not ($it | is-empty)
```

**Reply by panicbit** [reply]
Or even simpler:
```nu
['', "abc"] | where $it != ""
```
This one, however, would not filter out rows containing `$nothing`

**Reply by dbofmmbt** [reply]
Using `where` greatly improves readability. Nice to learn that it works on lists too. Thanks!
